tensorboard/webapp/plugins/npmi/views/data_selection/metric_arithmetic/metric_arithmetic_element/metric_arithmetic_element_component.ts

```diff
@@ -0,0 +1,112 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Input,
+  Output,
+  EventEmitter,
+} from '@angular/core';
+import {Validators, FormControl, ValidationErrors} from '@angular/forms';
+
+@Component({
+  selector: 'metric-arithmetic-element-component',
+  templateUrl: './metric_arithmetic_element_component.ng.html',
+  styleUrls: ['./metric_arithmetic_element_component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MetricArithmeticElementComponent {
+  @Input() metric!: string;
+  @Input() filterValues!: {min: string; max: string};
+  @Output() onRemove = new EventEmitter<string>();
+  @Output() onFilterChange = new EventEmitter<{min: number; max: number}>();
+  focusMin = false;
+  focusMax = false;
+  minFilterValid = true;
+  maxFilterValid = true;
+
+  readonly minFormControl = new FormControl(-1.0, [
+    Validators.required,
+    Validators.min(-1.0),
+    Validators.max(1.0),
+    this.minValueValidator.bind(this),
+  ]);
+
+  readonly maxFormControl = new FormControl(1.0, [
+    Validators.required,
+    Validators.min(-1.0),
+    Validators.max(1.0),
+    this.maxValueValidator.bind(this),
+  ]);
+
+  constructor() {
+    this.minFormControl.valueChanges.subscribe(() => {
```

Comment:
when subscribing something, you always want to unsubscribe with something like https://github.com/tensorflow/tensorboard/blob/master/tensorboard/webapp/settings/dialog_component.ts#L126.